### ISSUE TITLE
AbstractErrorWebExceptionHandler fails when the response has a non-standard status code

### ISSUE THREAD

**Issue by botcoder** [spring-projects/spring-boot#16691]
**Version**: spring-boot-2.1.4.RELEASE with spring-boot-starter-webflux

Providing custom error codes on the default web exception handler via `AbstractErrorWebExceptionHandler`, such as `ServerResponse.status(498)`, the error handling logic fails with `java.lang.IllegalArgumentException: No matching constant for [498]` when it attempts to log the exception:

```java
private void logError(ServerRequest request, ServerResponse response, Throwable throwable) {
// line 293 - statusCode lookup on HttpStatus enum
if (response.statusCode().equals(HttpStatus.INTERNAL_SERVER_ERROR)) {
```

`AbstractErrorWebExceptionHandler` should be able to handle "rawStatusCodes" besides the ones defined on HttpStatus enum, and/or return the response even if logging fails.

Related to: #15769, [SPR-16073](https://github.com/spring-projects/spring-framework/commit/9623cdec4883e134c2691cec56c39e59e1b4b80a#diff-e85b8359b4eaa4209148612aa0fbd3beR309)

**Comment by wilkinsona** [comment]
It think things would benefit from some changes in Framework. There's no way for us to get the raw status code. The alternative is to wrap `statusCode()` in a try-catch for an undocumented exception and infer from it that the status code wasn't known to `HttpStatus` which is a bit clunky. I've opened https://github.com/spring-projects/spring-framework/issues/22872.

**Comment by philwebb** [comment]
https://github.com/spring-projects/spring-framework/issues/22872 has now been fixed but it looks like a 5.2 only fix.
